listenbrainz/webserver/static/js/src/LastFMImporter.tsx

```diff
@@ -151,18 +150,24 @@ export default class LastFmImporter extends React.Component<
     }
   }
 
-  async getPage(page: number) {
-    /*
-     * Fetch page from Last.fm
-     */
+  /*
+   * @param {number} page - the page to fetch from Last.fm
+   * @param {number} retries - number times to retry in case of errors other than 40x
+   * Fetch page from Last.fm
+   */
+  async getPage(page: number, retries: number) {
     const { lastfmUsername } = this.state;
 
     const retry = (reason: string) => {
       // eslint-disable-next-line no-console
+      const timeout = 3000;
       console.warn(
-        `${reason} while fetching last.fm page=${page}, retrying in 3s`
+        `${reason} while fetching last.fm page=${page}, retrying in 
+        ${timeout / 1000}s`
       );
-      setTimeout(() => this.getPage(page), 3000);
+      if (retries > 0) {
+        setTimeout(() => this.getPage(page, retries - 1), timeout);
+      }
```

Comment:
Is there any indication to the user that we stopped retrying and that it's just not working, if we get to the max number of retries?

Reply:
There is some logic here https://github.com/metabrainz/listenbrainz-server/blob/a85eb2c7c003f91d14e12e016533f73cdde9def7/listenbrainz/webserver/static/js/src/LastFMImporter.tsx#L335, but I do not fully understand how this works. 

I think adding an else to this which raises an error could work?

Reply:
That would probably work.

The thrown error would be caught [here](https://github.com/metabrainz/listenbrainz-server/blob/a85eb2c7c003f91d14e12e016533f73cdde9def7/listenbrainz/webserver/static/js/src/LastFMImporter.tsx#L200), so you may need to modify the logic of that catch block to 
1. ensure we're not just retrying over and over and 
2. pass a clear error message that can be displayed to the end user

And finally, you'll want to wrap [the initial `getPage` call](https://github.com/metabrainz/listenbrainz-server/blob/a85eb2c7c003f91d14e12e016533f73cdde9def7/listenbrainz/webserver/static/js/src/LastFMImporter.tsx#L279) in a try-catch block and get the message to display in the catch.

Does that make sense to you?
I propose doing all this in another PR, to keep this current one from getting bigger.

Reply:
That makes sense. Sure, let's take that up in a different PR.

Reply:
I did it in here anyway since I was revamping the whole code anyway…